### PR TITLE
Add custom error support.

### PR DESCRIPTION
**Pull request by pongib**
I just see other PR with this custom error but not merge yet. So I decide to write my own custom error without remove any dependency and reserved error type as string not object like before PR.
Because I want to use this error message with throw option and should be simple as possible. 

If I do something wrong just clarify me. Thank you for your great work.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -67,9 +67,9 @@ function ratelimit(opts) {
     var delta = (limit.reset * 1000) - Date.now() | 0;
     var after = limit.reset - (Date.now() / 1000) | 0;
     this.set('Retry-After', after);
-
+
```
Comment (coderhaoxin) [inline index.js]: remove trailing spaces

File: index.js [hunk 2]
```diff
@@ -67,9 +67,9 @@ function ratelimit(opts) {
     var delta = (limit.reset * 1000) - Date.now() | 0;
     var after = limit.reset - (Date.now() / 1000) | 0;
     this.set('Retry-After', after);
-
+    
     this.status = 429;
-    this.body = 'Rate limit exceeded, retry in ' + ms(delta, { long: true });
+    this.body = opts.errMessage || 'Rate limit exceeded, retry in ' + ms(delta, { long: true }) + '.';
```
Comment (coderhaoxin) [inline index.js]: I prefer `errorMessage` ((^_^))

**Comment by coderhaoxin** [comment]
cc @dead-horse 

pre PR: https://github.com/koajs/ratelimit/pull/12

File: index.js (hunk 2 above)
Comment (pongib) [inline index.js]: Ok thank you. 😊

**Comment by pongib** [comment]
I just change errMessage to errorMessage that @coderhaoxin suggest and add docs for it.

**Comment by coderhaoxin** [comment]
land 4e38fb16c45c5bcc0782fc847471b18eaf35c502

**Comment by coderhaoxin** [comment]
`v2.4.0` published

**Pull request closed without merging**